presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/PartitionTable.java

```diff
@@ -164,10 +180,119 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
+        requireNonNull(constraint, "constraint is null");
+        checkArgument(tableHandle instanceof IcebergTableHandle, "tableHandle not class IcebergTableHandle, but instead %s", tableHandle.getClass().getName());
+        IcebergTableHandle icebergTableHandle = (IcebergTableHandle) tableHandle;
+
+        if (constraint.getSummary().isNone()) {
+            return empty();
+        }
+
+        List<PartitionField> partitionFields = icebergTable.spec().fields();
+        List<Type> icebergPartitionTypes = partitionTypes(partitionFields);
+        List<IcebergColumnHandle> columnHandles = getColumns(icebergTable.schema(), typeManager);
+        Map<Integer, IcebergColumnHandle> idToColumnHandle = columnHandles.stream()
+                .collect(toUnmodifiableMap(IcebergColumnHandle::getId, identity()));
+
+        ImmutableMap.Builder<Integer, ColumnFieldDetails> idToDetailsBuilder = ImmutableMap.builder();
+        for (int index = 0; index < partitionFields.size(); index++) {
+            PartitionField field = partitionFields.get(index);
+            Type type = icebergPartitionTypes.get(index);
+            idToDetailsBuilder.put(field.sourceId(), new ColumnFieldDetails(field, idToColumnHandle.get(field.sourceId()), type, toPrestoType(type, typeManager), type.typeId().javaClass()));
+        }
+        Map<Integer, ColumnFieldDetails> idToDetails = idToDetailsBuilder.build();
+
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            TableScan tableScan = getTableScan(session, TupleDomain.all(), icebergTableHandle.getSnapshotId(), icebergTable).includeColumnStats();
+            Map<StructLikeWrapper, Partition> partitions = getPartitions(tableScan);
+
+            Partition summingPartition = null;
+            for (Partition partition : partitions.values()) {
+                if (partitionMatches(partition, constraint, partitionFields, idToDetails)) {
+                    if (summingPartition == null) {
+                        summingPartition = partition;
+                        continue;
+                    }
+                    summingPartition.incrementFileCount(partition.getFileCount());
+                    summingPartition.incrementRecordCount(partition.getRecordCount());
+                    summingPartition.incrementSize(partition.getSize());
+                    summingPartition.updateSummingMin(partitionFields, partition.getMinValues(), partition.getNullCounts(), partition.getRecordCount());
+                    summingPartition.updateSummingMax(partitionFields, partition.getMaxValues(), partition.getNullCounts(), partition.getRecordCount());
+                    summingPartition.updateNullCount(partition.getNullCounts());
+                    summingPartition.updateColumnSizes(partition.getColumnSizes());
+                }
+            }
+
+            if (summingPartition == null) {
+                return empty();
```

Comment:
Don't static import `empty()` as the name doesn't have enough context to be standalone.

Reply:
Refilled.